common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -198,6 +198,15 @@ def delete_item(self, location, user_id, revision=None, **kwargs):
                 branches_to_delete = [ModuleStoreEnum.BranchName.published, ModuleStoreEnum.BranchName.draft]
             elif revision is None:
                 branches_to_delete = [ModuleStoreEnum.BranchName.draft]
+                draft_location = location.for_branch(ModuleStoreEnum.BranchName.draft)
+                try:
+                    item = self.get_item(draft_location)
+                    if getattr(item, 'has_children', False):
+                        # If item have has_published_version then delete published children also.
+                        if self.has_published_version(item):
+                            branches_to_delete.insert(0, ModuleStoreEnum.BranchName.published)
+                except ItemNotFoundError:
```

Comment:
<div id="summary">
     <h1>ItemNotFoundError at /settings/details/course-v1:test_org+249682352228747607428786230134803255090+test_run</h1>
     <pre class="exception_value">BlockKey(type=&#39;about&#39;, id=&#39;syllabus&#39;)</pre>
     <table class="meta">
   
       <tr>
         <th>Request Method:</th>
         <td>POST</td>
       </tr>
       <tr>
         <th>Request URL:</th>
         <td>http://localhost:8031/settings/details/course-v1:test_org+249682352228747607428786230134803255090+test_run</td>
       </tr>
   
       <tr>
         <th>Django Version:</th>
         <td>1.4.22</td>
       </tr>
   
       <tr>
         <th>Exception Type:</th>
         <td>ItemNotFoundError</td>
       </tr>
   
   
       <tr>
         <th>Exception Value:</th>
         <td><pre>BlockKey(type=&#39;about&#39;, id=&#39;syllabus&#39;)</pre></td>
       </tr>
   
   
       <tr>
         <th>Exception Location:</th>
         <td>/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/split_mongo/caching_descriptor_system.py in get_module_data, line 148</td>
       </tr>